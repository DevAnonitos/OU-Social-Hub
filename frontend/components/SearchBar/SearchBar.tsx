"use-client";

import React from 'react';
import { Input } from '../ui/input';

const SearchBar = () => {
  return (
    <div className='hidden flex-center rounded-full min-h-[54px] px-4 py-2 lg:flex'>
      <Input
        placeholder='Search your favorite Event ...' 
        className='searchbar-input placeholder:text-grey-500 focus:border-[1px] focus-visible:ring-0 focus-visible:ring-offset-0'
      />
    </div>
  );
};

export default SearchBar;