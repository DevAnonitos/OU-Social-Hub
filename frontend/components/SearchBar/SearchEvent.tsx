"use client";
import React from 'react'
import { Input } from '../ui/input';
import Image from 'next/image';
import { useRouter, useSearchParams } from 'next/navigation';

const SearchEvent = () => {
  return (
    <div className='flex items-center min-h-[54px] w-full overflow-hidden rounded-full bg-grey-50 px-4 py-2'>
      <Image 
        src="/assets/icons/search.svg" 
        alt="search" 
        width={24} 
        height={24} 
        priority
        quality={85}
      />
      <Input 
        type="text"
        placeholder={"Search..."}
        className="border-0 bg-grey-50 outline-offset-0 placeholder:text-grey-500 focus:border-0 focus-visible:ring-0 focus-visible:ring-offset-0"
      />
    </div>
  )
}

export default SearchEvent