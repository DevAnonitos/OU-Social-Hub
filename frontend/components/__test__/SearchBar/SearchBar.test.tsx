import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import SearchBar from "@/components/SearchBar/SearchBar";

describe("LeftSideBar", () => {
  it("Should render import LeftSideBar", () => {
    expect(SearchBar).toBeDefined()
    expect(SearchBar).toMatchSnapshot()
  })
});