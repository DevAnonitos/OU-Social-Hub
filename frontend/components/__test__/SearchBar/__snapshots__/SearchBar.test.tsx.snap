// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftSideBar Should render import LeftSideBar 1`] = `[Function]`;
